### PR TITLE
remove IKernelConnector

### PR DESCRIPTION
**Pull request by jonsequitur**

### CODE REVIEW

**Review by colombod — APPROVED** [review]

File: src/Microsoft.DotNet.Interactive.SQLite/ConnectSQLiteCommand.cs [hunk 1]
```diff
@@ -25,9 +25,9 @@ public ConnectSQLiteCommand()
         InvocationContext commandLineContext)
     {
         var connectionString = commandLineContext.ParseResult.GetValueForArgument(ConnectionStringArgument);
-        var connector = new SQLiteKernelConnector(connectionString);
         var localName = commandLineContext.ParseResult.GetValueForOption(KernelNameOption);
-        var kernel = await connector.CreateKernelAsync(localName);
+        var kernel1 = new SQLiteKernel($"sql-{localName}", connectionString);
+        var kernel = await Task.FromResult<Kernel>(kernel1);
         return new[] { kernel };
```
Comment (shyamnamboodiripad) [inline src/Microsoft.DotNet.Interactive.SQLite/ConnectSQLiteCommand.cs:31]: The await is a bit confusing. Why not remove the `async` keyword and say something like `return Task.FromResult(new [] { new SQLiteKernel($"sql-{localName}", connectionString) });`?

File: src/Microsoft.DotNet.Interactive.Tests/NamedPipeConnectionTests.cs [hunk 2]
```diff
@@ -23,15 +23,15 @@ public NamedPipeConnectionTests(ITestOutputHelper output) : base(output)
...
+    protected override Func<string, Task<ProxyKernel>> CreateConnector()
     {
         CreateRemoteKernelTopology(_pipeName);
 
         var connector = new NamedPipeKernelConnector(_pipeName);
 
         RegisterForDisposal(connector);
 
-        return connector;
+        return name => connector.CreateKernelAsync(name);
```
Comment (shyamnamboodiripad) [inline src/Microsoft.DotNet.Interactive.Tests/NamedPipeConnectionTests.cs:34]: Can this be simplified to `return connector.CreateKernelAsync`?

File: src/Microsoft.DotNet.Interactive.Tests/ProxyKernelConnectionTestsBase.cs [hunk 3]
```diff
@@ -133,7 +133,7 @@ public async Task fast_path_commands_over_proxy_can_be_handled()
     {
         var connector = CreateConnector();
```
Comment (shyamnamboodiripad) [inline src/Microsoft.DotNet.Interactive.Tests/ProxyKernelConnectionTestsBase.cs:134]: minor nit: Also rename this variable from `connector` to `createKernel` to match the above test?

File: src/dotnet-interactive/Connection/StdIoKernelConnector.cs [hunk 4]
```diff
@@ -60,17 +60,6 @@ public class StdIoKernelConnector : IKernelConnector
         _remoteKernelInfoCache = new ConcurrentDictionary<string, KernelInfo>();
     }
 
-    /// <remarks>
-    /// TODO: Does it even make sense to implement <see cref="IKernelConnector"/> here considering that we have
-    /// concepts (such as a single root <see cref="ProxyKernel"/>, and zero or more (child) <see cref="ProxyKernel"/>s)
-    /// that the <see cref="IKernelConnector"/> abstraction does not understand / support? Should the '#!connect stdio'
```
Comment (shyamnamboodiripad) [inline src/dotnet-interactive/Connection/StdIoKernelConnector.cs:66]: Just curious - any changes in our thoughts around the `#!connect stdio`? Can we remove that too?

**Review by shyamnamboodiripad — APPROVED** [review]

File: src/Microsoft.DotNet.Interactive.SQLite/ConnectSQLiteCommand.cs (hunk 1 above)
Comment (jonsequitur) [reply src/Microsoft.DotNet.Interactive.SQLite/ConnectSQLiteCommand.cs:31]: That was a weird artifact of the inlining refactoring. Fixed.

File: src/dotnet-interactive/Connection/StdIoKernelConnector.cs (hunk 4 above)
Comment (jonsequitur) [reply src/dotnet-interactive/Connection/StdIoKernelConnector.cs:66]: I think we might still want this, especially as project integration comes along and we think about .NET Interactive instances for other TFMs, different dependency graphs, etc.

**Pull request merged**